dig.go

```diff
@@ -264,123 +271,6 @@ func (c *Container) isAcyclic(n *node) error {
 	return detectCycles(n.Params, c.nodes, []key{n.key})
 }
 
-// Retrieve a type from the container
-func (c *Container) get(e edge) (reflect.Value, error) {
-	if v, ok := c.cache[e.key]; ok {
-		return v, nil
-	}
-
-	if IsIn(e.t) {
-		// We do not want parameter objects to be cached.
-		return c.createInObject(e.t)
-	}
-	if embedsType(e.t, _inPtrType) {
-		return _noValue, fmt.Errorf(
-			"%v embeds *dig.In which is not supported, embed dig.In value instead", e.t,
-		)
-	}
-
-	if e.t.Kind() == reflect.Ptr {
-		if IsIn(e.t.Elem()) {
-			return _noValue, fmt.Errorf(
-				"dependency %v is a pointer to dig.In, use value type instead", e.t,
```

Comment:
These checks are now part of newParam. So, they're done on `Provide` rather than `Invoke`.